addons/l10n_jo_edi/tests/test_jo_edi_precision.py

```diff
@@ -0,0 +1,164 @@
+from odoo.tests import tagged
+from odoo.tools.float_utils import float_is_zero
+from odoo.addons.l10n_jo_edi.tests.jo_edi_common import JoEdiCommon
+
+JO_MAX_DP = 9
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestJoEdiPrecision(JoEdiCommon):
+    def _equality_check(self, vals_dict, up_to_jo_max_dp=True):
+        def equal_strict(val1, val2):
+            return val1 == val2
+
+        def equal_jo_max_dp(val1, val2):
+            return float_is_zero(val1 - val2, JO_MAX_DP)
+
+        equals = equal_jo_max_dp if up_to_jo_max_dp else equal_strict
+        first_tuple = None
+        error_message = ""
+        for label, value in vals_dict.items():
+            if not first_tuple:
+                first_tuple = (label, value)
+            else:
+                if not equals(value, first_tuple[1]):
+                    error_message += f"{label} ({value}) != {first_tuple[0]} ({first_tuple[1]})\n"
+        return error_message
+
+    def _extract_vals_from_subtotals(self, subtotals, defaults):
+        for subtotal in subtotals:
+            tax_percent = float(subtotal.findtext('{*}TaxCategory/{*}Percent', default=-1))
+            if tax_percent == -1:  # special (fixed amount) tax
+                defaults.update({
+                    'taxable_amount_special': float(subtotal.findtext('{*}TaxableAmount')),
+                    'tax_amount_special': float(subtotal.findtext('{*}TaxAmount')),
+                })
+                defaults['total_tax_amount'] += defaults['tax_amount_special']
+            else:
+                defaults.update({
+                    'taxable_amount_general': float(subtotal.findtext('{*}TaxableAmount')),
+                    'tax_amount_general_subtotal': float(subtotal.findtext('{*}TaxAmount')),
+                    'tax_percent': tax_percent / 100,
+                })
+                defaults['total_tax_amount'] += defaults['tax_amount_general_subtotal']
+
+        return defaults
+
+    def validate_jo_edi_numbers(self, xml_string):
```

Comment:
I would put the long docstring of the other method here.  (and refer in the method to this test)